sdk/core/azure-core/src/test/java/com/azure/core/util/polling/PollerTests.java

```diff
@@ -423,6 +429,75 @@ public void getResultCanBeCalledFromOperatorChainTest() {
         pollingContext.getLatestResponse().equals(response2);
     }
 
+    @Test
+    public void verifyExceptionPropagationFromPollingOperation() {
+        final Response activationResponse = new Response("Foo");
+        when(activationOperation.apply(any()))
+            .thenReturn(Mono.defer(() -> Mono.just(activationResponse)));
+
+        final AtomicReference<Integer> cnt = new AtomicReference<>(0);
```

Comment:
Why `AtomicReference` over `AtomicInteger`?

Reply:
We could use either..